osgp/protocol-adapter-oslp/web-device-simulator/src/main/java/org/opensmartgridplatform/webdevicesimulator/service/OslpChannelHandler.java

```diff
@@ -95,135 +95,344 @@
+        return Oslp.Message.newBuilder()
+                .setSetScheduleResponse(SetScheduleResponse.newBuilder().setStatus(Oslp.Status.OK))
+                .build();
+    }
+
+    private static Message createGetActualPowerUsageResponse() {
+        // yyyyMMddhhmmss z
+        final SimpleDateFormat utcTimeFormat = new SimpleDateFormat("yyyyMMddHHmmss");
+        utcTimeFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+
+        final Date currentDateTime = new Date();
+        final String utcTimestamp = utcTimeFormat.format(currentDateTime);
+
+        @SuppressWarnings("deprecation")
+        final int actualConsumedPower = currentDateTime.getMinutes();
+
+        return Oslp.Message.newBuilder()
+                .setGetActualPowerUsageResponse(GetActualPowerUsageResponse.newBuilder()
+                        .setPowerUsageData(PowerUsageData.newBuilder()
+                                .setRecordTime(utcTimestamp)
+                                .setMeterType(MeterType.P1)
+                                .setTotalConsumedEnergy(actualConsumedPower * 2L)
+                                .setActualConsumedPower(actualConsumedPower)
+                                .setPsldData(PsldData.newBuilder().setTotalLightingHours(actualConsumedPower * 3))
+                                .setSsldData(SsldData.newBuilder()
+                                        .setActualCurrent1(1)
+                                        .setActualCurrent2(2)
+                                        .setActualCurrent3(3)
+                                        .setActualPower1(1)
+                                        .setActualPower2(2)
+                                        .setActualPower3(3)
+                                        .setAveragePowerFactor1(1)
+                                        .setAveragePowerFactor2(2)
+                                        .setAveragePowerFactor3(3)
+                                        .addRelayData(Oslp.RelayData.newBuilder()
+                                                .setIndex(ByteString.copyFrom(new byte[] { 2 }))
+                                                .setTotalLightingMinutes(480))
+                                        .addRelayData(Oslp.RelayData.newBuilder()
+                                                .setIndex(ByteString.copyFrom(new byte[] { 3 }))
+                                                .setTotalLightingMinutes(480))
+                                        .addRelayData(Oslp.RelayData.newBuilder()
+                                                .setIndex(ByteString.copyFrom(new byte[] { 4 }))
+                                                .setTotalLightingMinutes(480))))
+                        .setStatus(Oslp.Status.OK))
+                .build();
+    }
+
+    private static Message createGetPowerUsageHistoryWithDatesResponse(
+            final GetPowerUsageHistoryRequest powerUsageHistoryRequest) throws ParseException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.text.ParseException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

Reply:
It's a bit hard to tell what has changed, and what has been moved around.
Jenkins makes sense here though, no need to declare exceptions that cannot be thrown.

Reply:
I removed the throw declaration which will resolve the SonarQube issue.